aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/S3ClientOptions.java

```diff
@@ -103,4 +143,22 @@ public S3ClientOptions withPathStyleAccess(boolean pathStyleAccess) {
       return this;
     }
 
-}
\ No newline at end of file
+    /**
+     * <p>
+     * Configures the client to disbale chunked encoding for all requests.
+     * </p>
+     * <p>
+     * The default behaviour is to detect API request which enables chunked
+     * encoding automatically for PutObjectRequest and UploadPartRequest.
+     * Setting this flag will result in disabling chunked encoding for all
+     * requests.
+     * </p>
+     * @return The updated S3ClientOptions object with chunked encoding
+     *         disabled setting.
+     */
+    public S3ClientOptions disableChunkedEncoding() {
```

Comment:
@manikandanrs suggested we clearly call out in the documentation that this has performance implications when using the real S3. Could you add a blurb saying this is not recommended for typical use cases and should only be used when the S3 implementation doesn't support chunked encoding?